src/Core/Tools/SendFeatures/Services/SendAuthorizationService.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Platform.Push;
+using Bit.Core.Tools.Entities;
+using Bit.Core.Tools.Enums;
+using Bit.Core.Tools.Models.Business;
+using Bit.Core.Tools.Repositories;
+using Microsoft.AspNetCore.Identity;
+
+namespace Bit.Core.Tools.Services;
+
+public class SendAuthorizationService : ISendAuthorizationService
+{
+    private readonly ISendRepository _sendRepository;
+    private readonly IPasswordHasher<User> _passwordHasher;
+    private readonly IPushNotificationService _pushNotificationService;
+    private readonly IReferenceEventService _referenceEventService;
+    private readonly ICurrentContext _currentContext;
+
+    public SendAuthorizationService(
+        ISendRepository sendRepository,
+        IPasswordHasher<User> passwordHasher,
+        IPushNotificationService pushNotificationService,
+        IReferenceEventService referenceEventService,
+        ICurrentContext currentContext)
+    {
+        _sendRepository = sendRepository;
+        _passwordHasher = passwordHasher;
+        _pushNotificationService = pushNotificationService;
+        _referenceEventService = referenceEventService;
+        _currentContext = currentContext;
+    }
+
+    public (bool grant, bool passwordRequiredError, bool passwordInvalidError) SendCanBeAccessed(Send send,
```

Comment:
🤔 Is there a higher-level type you could output here? The fields appear to be mutually exclusive, which leads me to believe this should be an `Enum`.

Reply:
This hasn't been addressed yet.

Reply:
Fixed this